tests/integration_tests/functional/test_snapshot_advanced.py

```diff
@@ -25,106 +19,79 @@
 scratch_drives = ["vdb", "vdc", "vdd", "vde", "vdf"]
 
 
-def firecracker_artifacts(*args, **kwargs):
-    """Return all available firecracker binaries."""
-    artifacts = ArtifactCollection(_test_images_s3_bucket())
-    # Fetch all firecracker binaries.
-    return artifacts.firecrackers(*args, **kwargs)
-
-
-def firecracker_id(fc):
-    """Render a nice ID for pytest parametrize."""
-    if isinstance(fc, FirecrackerArtifact):
-        return f"firecracker-{fc.version}"
-    return None
-
-
-@pytest.mark.parametrize(
-    "firecracker",
-    firecracker_artifacts(
-        min_version="1.1.0",
-        max_version=get_firecracker_version_from_toml(),
-    ),
-    ids=firecracker_id,
-)
-def test_restore_old_snapshot(bin_cloner_path, firecracker):
+def test_restore_old_to_current(bin_cloner_path, firecracker_release):
     """
-    Restore from snapshots obtained with previous versions of Firecracker.
+    Restore snapshots from previous supported versions of Firecracker.
+
+    For each firecracker release:
+    1. Snapshot with the past release
+    2. Restore with the current build
 
     @type: functional
     """
+
+    # due to ARM bug fixed in commit 822009ce
+    if platform.machine() == "aarch64" and firecracker_release.version_tuple < (
+        1,
+        1,
+        4,
+    ):
```

Comment:
Since we're on the topic of relaxing line length limits, black's formatting can be quite interesting (see [here](https://github.com/firecracker-microvm/firecracker/blob/4d4f5aa94a5b0ee84b3f7e24053eb8c6382b1d77/tests/integration_tests/build/test_dependencies.py#L56-L58) for another example). Is there anything we can do about that? 😅

Reply:
Yeah this looks awful